source/component/PasDoc_Parser.pas

```diff
@@ -898,20 +898,36 @@ procedure TParser.ParseRoutine(out M: TPasRoutine;
     until IsSemicolon and (Level = 0);
   end;
 
+  function IsParameterIdentifier(Token: TToken): Boolean;
+  var
+    Text: string;
+  begin
+    if Token.MyType = TOK_IDENTIFIER then
+      Result := True
+    else if Token.MyType = TOK_KEYWORD then begin
+      Text := AnsiLowerCase(Token.Data);
+      Result := (Text = 'string') or (Text = 'array') or (Text = 'set') or (Text = 'of') or (Text = 'file');
```

Comment:
No need to compare text here, instead use ready `Token.KeyWord`. So this is better:

```pascal
Result := Token.KeyWord in [KEY_STRING, KEY_ARRAY, KEY_SET, ...];
```

( I see previous code also used `AnsiLowerCase(T.Data) = 'string')`,I didn't catch it earlier, sorry. )

Reply:
I didn't realise that functionality existed, changed :)